force-app/main/default/classes/SOQL.cls

```diff
@@ -85,20 +91,32 @@ public with sharing class SOQL {
      * @return A map of Task Ids to Task records.
      * Update the corresponding test class and testGetTasksDueNextWeek method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
-
+    public static Map <Id, Task> tasksDueNextWeek() {
+        Map <Id, Task> taskMap = new Map <Id, Task>();
+        List <Task> taskList = [SELECT Id FROM Task WHERE ActivityDate <= NEXT_N_DAYS:7];
+        for(Task t : taskList){
+            taskMap.put(t.Id, t);
+        }
+        return taskMap;
```

Comment:
There is a special constructor for Map<Id, SObject> that takes a List and separates it into a map of the Id to the record itself. 

Great use of the key words for NEXT_N_DAYS:n as well!